packages/webpack/src/plugin.js

```diff
@@ -0,0 +1,362 @@
+
+        // This part adds LavaMoat runtime to webpack runtime for every chunk that needs runtime.
+        // I stole this from Zach of module federation fame
+
+        const onceForChunkSet = new WeakSet()
+        const runtimeOptions = {
+          lockdown: options.lockdown,
+        }
+
+        // Define a handler function to be called for each chunk in the compilation.
+        compilation.hooks.additionalChunkRuntimeRequirements.tap(
+          PLUGIN_NAME + '_runtime',
+          (chunk /*, set*/) => {
+            if (chunk.hasRuntime()) {
+              if (!PROGRESS.done('gneratorCalled')) {
+                mainCompilationWarnings.push(
+                  new WebpackError(
+                    'LavaMoatPlugin: Something was generating runtime before all modules were identified. This might be part of a sub-compilation of a plugin. Please check for any unwanted interference between plugins.',
+                  ),
+                )
+                diag.rawDebug(
+                  1,
+                  '> skipped adding runtime (additionalChunkRuntimeRequirements)',
+                )
+                // It's possible to generate the runtime with an empty policy to make the wrapped code work.
+                // It's no longer necessasry now that `generate` function is only wrapping anything if paths were processed,
+                // which corresponds to it being the main compilation. But plugins may exist that conflict with that assumption;
+                // in which case we're gonna have to bring back the runtime with empty policy
+              } else {
+                diag.rawDebug(
+                  1,
+                  '> adding runtime (additionalChunkRuntimeRequirements)',
+                )
+                // narrow down the policy and map to module identifiers
+                const policyData = identifierLookup.getTranslatedPolicy()
+
+                const lavaMoatRuntime = assembleRuntime(RUNTIME_KEY, [
+                  {
+                    name: 'root',
+                    data: identifierLookup.root || null,
+                    json: true,
+                  },
+                  {
+                    name: 'idmap',
+                    data: identifierLookup.identifiersForModuleIds || null,
+                    json: true,
+                  },
+                  {
+                    name: 'unenforceable',
+                    data: identifierLookup.unenforceableModuleIds || null,
+                    json: true,
+                  },
+                  { name: 'options', data: runtimeOptions, json: true },
+                  { name: 'policy', data: policyData, json: true },
+                  { name: 'ENUM', file: path.join(__dirname, './ENUM.json'), json: true },
+                  {
+                    name: 'endowmentsToolkit',
+                    shimRequire:
+                      'lavamoat-core/src/endowmentsToolkit.js',
```

Comment:
is there another way to do this that doesn't require reaching into the source this way?